org.eclipse.jdt.debug/model/org/eclipse/jdt/internal/debug/core/model/JDIDebugTarget.java

```diff
@@ -1781,8 +1784,9 @@ protected void suspendedByBreakpoint(JavaBreakpoint breakpoint,
 	 */
 	protected void cancelSuspendByBreakpoint(JavaBreakpoint breakpoint)
 			throws DebugException {
-		setSuspended(false);
-		resumeThreads();
+		if (suspended.getAndSet(false)) {
```

Comment:
Yes, ```suspended``` is left-over from previous code. It should be ```fSuspended```.

There is ```SuspendVMBreakpointsTests``` but I guess no test there checks whether this method works correctly (in particular, checking whether calling this method twice works on the same debug target).

I can try to add a test for it. In this issue, or in a new one? No idea how long it will take to write the test.